src/components/chat/messages/ChatMessage.vue

```diff
@@ -10,56 +8,59 @@
 
     <!-- Delete Dialog -->
     <q-dialog v-model="deleteDialog">
-      <delete-message-dialog :address="address" :payloadDigest="payloadDigest" :index="index" />
+      <delete-message-dialog :address="address" :payloadDigest="message.payloadDigest" :index="index" />
     </q-dialog>
 
+    <!-- Information tooltip -->
+    <q-tooltip>
+      <div class="col-auto q-pa-none">
+        <div class="row-auto">{{ stampPrice }}</div>
+        <div class="row-auto">{{ timestampString }}</div>
+      </div>
+    </q-tooltip>
+
     <chat-message-menu
       :address="address"
-      :payloadDigest="payloadDigest"
       :message="message"
-      :index="index"
       @txClick="transactionDialog = true"
       @deleteClick="deleteDialog = true"
-      @replyClick="replyClicked({ address, payloadDigest })"
+      @replyClick="replyClicked({ address, payloadDigest: message.payloadDigest })"
     />
-    <q-tooltip>
-      <div class="col-auto q-pa-none">
-        <div class="row-auto">{{stampPrice}}</div>
-        <div class="row-auto">{{timestampString}}</div>
-      </div>
-    </q-tooltip>
 
-    <div class="row">
-      <div class="col">
-        <!-- TODO: Assign random color based on seed from name -->
-        <div class="col text-weight-bold" style="border-bottom: 1px solid grey;" v-if="showHeader"><div :style="nameColor"> {{contact.name}} </div></div>
-        <chat-message-section :key="index" :items="message.items" :address="address" />
-      </div>
-      <div class="col-auto" style="min-width: 100px">
-        <div class="col text-weight-bold" style="border-bottom: 1px solid grey;" v-if="showHeader">&nbsp;</div>
-        <div class="row-auto">{{shortTimestamp}}</div>
-      </div>
-      <div v-if="message.status==='error'" class="row justify-end q-pt-xs" style="full-width">
-        <div class="col-auto">
-          <q-icon name="error" color="red" />
-        </div>
+    <div class='col'>
+      <div class = 'q-px-sm' v-for="(item, index) in message.items" :key="index" >
+        <chat-message-reply v-if="item.type=='reply'" :payloadDigest="item.payloadDigest" :address="address"/>
+        <chat-message-text v-else-if="item.type=='text'" :text="item.text" />
+        <chat-message-image v-else-if="item.type=='image'" :image="item.image" />
```

Comment:
An aside: FWIW, I'd tried to send a PDF and it didn't work.  Maybe a file entry with a MIME type or something would be nice?

Reply:
Yeah, the logic and entry here assumes an image. Shouldn't be too hard to generalize it here but there's a few choices to be made that we should talk over.